src/pk/ecc/ecc_ssh_ecdsa_encode_name.c

```diff
@@ -0,0 +1,57 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+#include "tomcrypt_private.h"
+
+/**
+   @file ecc_ssh_ecdsa_encode_name.c
+   Curve/OID to SSH+ECDSA name string mapping per RFC5656
+   Russ Williams
+*/
+
+int ecc_ssh_ecdsa_encode_name(char *buffer, unsigned long buflen, const ecc_key *key)
+{
+   int err;
+
+   /* Check for three named curves: nistp256, nistp384, nistp521 */
+   if (ecc_cmp_oid_str("1.2.840.10045.3.1.7", key) == CRYPT_OK) {
+      /* nistp256 - secp256r1 - OID 1.2.840.10045.3.1.7 */
+
+      if (buflen <= 19) { err = CRYPT_BUFFER_OVERFLOW; goto error; }
+      strcpy(buffer, "ecdsa-sha2-nistp256");
+   }
+   else if (ecc_cmp_oid_str("1.3.132.0.34", key) == CRYPT_OK) {
+      /* nistp384 - secp384r1 - OID 1.3.132.0.34 */
+
+      if (buflen <= 19) { err = CRYPT_BUFFER_OVERFLOW; goto error; }
+      strcpy(buffer, "ecdsa-sha2-nistp384");
+   }
+   else if (ecc_cmp_oid_str("1.3.132.0.35", key) == CRYPT_OK) {
+      /* nistp521 - secp521r1 - OID 1.3.132.0.35 */
+
+      if (buflen <= 19) { err = CRYPT_BUFFER_OVERFLOW; goto error; }
+      strcpy(buffer, "ecdsa-sha2-nistp512");
+   }
+   else { /* Otherwise, we use the OID of the curve */
+      char oidstr[32];
```

Comment:
Isn't it too small? I am for oidstr[64].

Reply:
Changed to 64